R/benchmark.R

```diff
@@ -148,4 +148,22 @@ default_params <- function(bm, ...) {
   out
 }
 
-
+#' Extract the parameter summary as a tibble
+#'
+#' Extract a tibble that provides the parameters used in a run and the
+#' error status
+#'
+#' @param run The object returned by either `run_benchmark` or `run_one`
+#' @return a tibble
+#' @export
+get_params_summary <- function(run) {
+  if (!inherits(run, "arrowbench_results")) {
+    stop("run objects need to be of class arrowbench_results")
+  }
+  purrr::map_df(seq_along(run), ~{
+    d <- run[[.x]]$params
+    d <- dplyr::as_tibble(d)
```

Comment:
I suspect that this should [`as.data.frame()`](https://github.com/ursacomputing/arrowbench/blob/4022c6e37a8aa2eebd579a8d710ee5e05c1c7372/R/result.R#L11-L15) so that you get one benchmark case per row. It looks like you get one benchmark case * packages used (+ maybe more?) right now.

Reply:
Good catch. I think we can just drop the packages here as I don't _think_ they are needed. But it also made me rethink returning a tibble. Simpler to maybe just use a data.frame.